crates/wast/src/ast/wast.rs

```diff
@@ -45,6 +45,40 @@ impl Peek for WastDirectiveToken {
     }
 }
 
+/// Enumerate the available types that can be checked in `AssertReturn*Nan`.
+#[derive(Debug, PartialEq, Eq, Hash, Copy, Clone)]
+#[allow(missing_docs)]
+pub enum WastCheckType {
+    F32x4, F64x2
+}
+
+impl Parse<'_> for WastCheckType {
+    fn parse(parser: Parser) -> Result<Self> {
+        let mut l = parser.lookahead1();
+        if l.peek::<kw::f32x4>() {
+            parser.parse::<kw::f32x4>()?;
+            Ok(WastCheckType::F32x4)
+        } else if l.peek::<kw::f64x2>() {
+            parser.parse::<kw::f64x2>()?;
+            Ok(WastCheckType::F64x2)
+        } else {
+            Err(l.error())
+        }
+    }
+}
+
+/// Helper [`Parse`] implementation to capture an optional [`WastCheckType`].
+impl<'a> Parse<'a> for Option<WastCheckType> {
```

Comment:
I think instead of adding this impl you can implement `Peek for WastCheckType`, and this impl will then get synthesized and that should also remove the need for an `RParen` peek as well!